src/app/cours/cours.component.ts

```diff
@@ -12,45 +14,155 @@ import { Unit } from '../model/unit';
+        console.error(err)}
+    ) 
   }
 
   deleteAll(): void {
-    //this.myClasses = this.coursService.deleteAllClasses();
-    this.myClasses = [];
+    this.myClasses = this.coursService.deleteAllClasses();
   }
 
-  deleteClass(id: number): void {
-      const currentClasses = this.myClasses.slice(0, this.myClasses.length);
-      this.myClasses = this.coursService.deleteClassById(currentClasses, id);
+  deleteClass(id: number, coursName: string): void {
+    const swalWithBootstrapButtons = Swal.mixin({
+      customClass: {
+        confirmButton: 'btn btn-success',
+        cancelButton: 'btn btn-danger'
+      },
+      buttonsStyling: false
+    })
+
+    swalWithBootstrapButtons.fire({
+      title: 'Confirmez-vous la suppression du cours '+ coursName +' ?',
+      text: "Cette action est irreversible!",
+      icon: 'warning',
+      showCancelButton: true,
+      confirmButtonText: 'Oui',
+      cancelButtonText: 'Non',
+      reverseButtons: true
+    }).then((result) => {
+      if (result.value) {
+        this.coursService.deleteClassById(id)
+        .subscribe(
+          (data) => {
+            let oldClasses = this.myClasses.slice(0, this.myClasses.length);
+            const idClass = oldClasses.findIndex(c => c.id === id);
+            if(idClass !== -1){
+              oldClasses.splice(idClass, 1); 
+              this.myClasses = oldClasses;
+              swalWithBootstrapButtons.fire(
+                'Suppression',
+                'Le cours '+ coursName + ' à bien été supprimé',
+                'success'
+              )
+            } else {
+              console.error('error id');
+            }  
+          },
+          (err)=> { console.error(err)}
+        );
+       
+      } else if (
+        result.dismiss === Swal.DismissReason.cancel
+      ) {
+        swalWithBootstrapButtons.fire(
+          'Annulation',
+          'Suppresion interrompue',
+          'error'
+        )
+      }
+    })
+
+
+
+
+
+
+
+    this.coursService.deleteClassById(id)
+    .subscribe(
+      (data) => {
+        let oldClasses = this.myClasses.slice(0, this.myClasses.length);
+        const idClass = oldClasses.findIndex(c => c.id === id);
+        if(idClass !== -1){
+          oldClasses.splice(idClass, 1); 
+          this.myClasses = oldClasses;
+        } else {
+          console.error('error id');
+        } 
+      },
+      (err)=> { console.error(err)}
+    );
   }
 
+
   onSubmit(): void {
+    const form = this.classForm.value;
     if (this.classForm.invalid) {
-      alert('Vous n\'avez pas remplis tous les champs');
-      return;
+      Swal.fire(
+        'Attention',
+        'Vous n\'avez pas remplis tous les champs',
+        'warning'  
+      );
+    }
+    else {
+      form.label = form.label.charAt(0).toUpperCase() + form.label.slice(1);
+      form.professor = this.profSelect;
+
+      this.coursService.postClass(form)
+      .subscribe(
+        (data)=> { 
+          this.getCours();
```

Comment:
you don't need to call this, just add the new class into this.myClasses

Reply:
Ok but how can I manage the item's id number ?